### PR TITLE
feat: UI implementation for Linting and Validation

### LINKED ISSUE #1499: [linting & validation] UI implementation for Linting and Validation
**Is your feature request related to a problem? Please describe.**
Need to align with the design

**Describe the solution you'd like**
https://www.figma.com/file/Ux0vucOvaLL2xH22WpFgV65X/MarcB_Composer-UI?node-id=3268%3A25239

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by lei9444**
## Description

UI implementation for Linting and Validation
closes #1499
## Task Item
closes https://github.com/microsoft/BotFramework-Composer/issues/1499

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code refactor (non-breaking change which improve code quality, clean up, add tests, etc)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Doc update (document update)

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have functionally tested my change

### CODE REVIEW

File: Composer/packages/client/src/App.tsx [hunk 1]
```diff
@@ -71,6 +72,14 @@ const topLinks = (botLoaded: boolean) => {
       exact: false,
       underTest: true, // will delete
     },
+    {
+      to: '/notifications',
+      iconName: 'Warning',
+      labelName: formatMessage('Notifications'),
+      activeIfUrlContains: '/notifications',
```
Comment (cwhitten) [inline Composer/packages/client/src/App.tsx]: where did this `activeIfUrlContains` convention come from?

File: Composer/packages/client/src/pages/notifications/NotificationHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+// Licensed under the MIT License.
+
+import * as React from 'react';
+import { Dropdown, IDropdownStyles } from 'office-ui-fabric-react/lib/Dropdown';
+import formatMessage from 'format-message';
+import { useMemo } from 'react';
+
+import { notificationHeader, notificationHeaderText } from './styles';
+
+const dropdownStyles: Partial<IDropdownStyles> = {
```
Comment (cwhitten) [inline Composer/packages/client/src/pages/notifications/NotificationHeader.tsx]: please move this to the `styles` file for this component

File: Composer/packages/client/src/pages/notifications/NotificationList.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import { DetailsList, DetailsListLayoutMode, SelectionMode, IColumn, FontIcon } from 'office-ui-fabric-react';
+
+import { INotification } from './types';
+import { notification, typeIcon, listRoot } from './styles';
+
+export interface INotificationListProps {
+  items: INotification[];
+}
+
+const icons = {
```
Comment (cwhitten) [inline Composer/packages/client/src/pages/notifications/NotificationList.tsx]: please move this to the `styles` file

File: Composer/packages/client/src/pages/notifications/NotificationList.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  Error: { iconName: 'ErrorBadge', color: '#A80000', background: '#FED9CC' },
+  Warning: { iconName: 'Warning', color: '#8A8780', background: '#FFF4CE' },
+};
+
+const columns: IColumn[] = [
+  {
+    key: 'column1',
```
Comment (cwhitten) [inline Composer/packages/client/src/pages/notifications/NotificationList.tsx]: Please use this field to give better descriptions for these columns

File: Composer/packages/client/src/pages/notifications/useNotifications.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import { INotification } from './types';
+
+const DiagnosticSeverity = ['Error', 'Warning', 'Information', 'Hint'];
+
+export default function useNotifications() {
+  const { state } = useContext(StoreContext);
+  const [filter, setFilter] = useState('All');
+  const { dialogs, luFiles, lgFiles } = state;
+
+  const memorized = useMemo(() => {
```
Comment (cwhitten) [inline Composer/packages/client/src/pages/notifications/useNotifications.tsx]: s/memorized/memoized

File: Composer/packages/client/src/pages/notifications/useNotifications.tsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+import { StoreContext } from '../../store';
+import { createSingleMessage } from '../../utils/lgUtil';
+
+import { INotification } from './types';
+
+const DiagnosticSeverity = ['Error', 'Warning', 'Information', 'Hint'];
+
+export default function useNotifications() {
+  const { state } = useContext(StoreContext);
+  const [filter, setFilter] = useState('All');
```
Comment (cwhitten) [inline Composer/packages/client/src/pages/notifications/useNotifications.tsx]: What are the different filters that can be set other than the string 'All' ? We should hoist these out and give it a type for better self-documentation if it's more like an enum.

File: Composer/packages/client/src/pages/notifications/useNotifications.tsx [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+          type: DiagnosticSeverity[diagnostic.Severity],
+          location,
+          message: createSingleMessage(diagnostic),
+        });
+      });
+    });
+    return { notifactions, locations: [...locations] };
+  }, [dialogs, luFiles, lgFiles]);
+
+  const notifications: INotification[] = memorized.notifactions.filter(x => filter === 'All' || x.location === filter);
```
Comment (cwhitten) [inline Composer/packages/client/src/pages/notifications/useNotifications.tsx]: see comment above, 'All' seems like a magic string

File: Composer/packages/client/tsconfig.json [hunk 8]
```diff
@@ -3,7 +3,7 @@
     "outDir": "./dist/",
     "noImplicitAny": false,
     "module": "es6",
-    "target": "es5",
+    "target": "es6",
```
Comment (cwhitten) [inline Composer/packages/client/tsconfig.json]: why are you making this change?

File: Composer/packages/client/src/App.tsx (hunk 1 above)
Comment (lei9444) [reply Composer/packages/client/src/App.tsx]: I found that the activeIfUrlContains was not useful for the component. I have removed it now.

**Comment by lei9444** [comment]
@cwhitten @a-b-r-o-w-n @boydc2014 In this PR, The notification page just shows all files' parse error. Do we need to implement that when clicking the error item the page jump to the specified location? If yes, I will do it in the next PR. I think this PR can be merged first.

**Comment by lei9444** [comment]
@cwhitten @a-b-r-o-w-n @boydc2014 I have fixed the conflicts, Do you have any ideas about this PR? If it's ok, I think we can merge it first.

**Comment by boydc2014** [comment]
Looks good, @cwhitten do you think this is good to go?

File: Composer/packages/client/src/pages/notifications/NotificationHeader.tsx [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
+import { Dropdown, IDropdownOption } from 'office-ui-fabric-react/lib/Dropdown';
+import formatMessage from 'format-message';
+import { useMemo } from 'react';
+
+import { notificationHeader, notificationHeaderText, dropdownStyles } from './styles';
+
+const createOptions = (items: string[]): IDropdownOption[] => {
+  const defaultOptions: IDropdownOption[] = [{ key: 'Show All Locations', text: 'All', data: '', isSelected: true }];
```
Comment (cwhitten) [inline Composer/packages/client/src/pages/notifications/NotificationHeader.tsx]: 'Show all Locations'  and 'All' should be wrapped in `formatMessage`

**Comment by lei9444** [comment]
> @lei9444 can you add some tests for this change please? thanks.

@cwhitten, I have added some unit tests and e2e tests. The diagnostics of lu and dialog does not work now, I will add them to e2e test after they are ok.

**Pull request merged**